bokeh/util/serialization.py

```diff
@@ -112,8 +116,14 @@ def traverse_data(datum, is_numpy=is_numpy, use_numpy=True):
         use_numpy: toggle numpy as a dependency for testing purposes
     """
     is_numpy = is_numpy and use_numpy
-    if is_numpy and not any(isinstance(el, (list, tuple)) for el in datum):
-        return transform_array(np.asarray(datum))
```

Comment:
I also think this was a bug that just happened to "work" because we usually have a list with just one array, or all the same size. Different sized array, (i.e. different sized images) would behave badly:
```
In [7]: np.asarray(c)
Out[7]:
array([array([1, 2, 3]), array([[1, 2],
       [3, 4]])], dtype=object)
```